### ISSUE TITLE
No Devices are dicoverd after a Echo Dot reset

### ISSUE THREAD

**Issue by altheeagle** [bwssytems/ha-bridge#1185]
Since yesterday I had a Problem with my Echo Dot v2, it stopped working (can't connect to my Wifi). After I resetted my Echo Dot and got it back running I tried to (re-)disover my devices and nothing has been found. The Configuration of my HA-Bridge Setup hasn't been changed (by me). I tried all steps in the troubleshooting section in the FAQ, but still can't find any of my Devices. In the Log (in the Webgui or in the journalctl) the is no trace of any connection attempts from my Echo Dot or any other Device
[habridge-log.txt](https://github.com/bwssytems/ha-bridge/files/4303167/habridge-log.txt)

**Comment by bwssytems** [comment]
Trace upnp is not turned on for that log. Please turn that on, do a discover, then post the log.

**Comment by altheeagle** [comment]
Thank you for the fast Response. I've thought that I had "Trace UPNP Calls" set to true, but it was set to false. So I changed it to true on the "Bridge Control" Tab und hit Save, but it didn't save the Setting, it was set to false again. Did I something wrong?

**Comment by bwssytems** [comment]
Not sure about that not sticking. Do you see any errors in your log after you hit save that it cannot save to your config file? In your other log it shows that it is loading from a path in your home directory, is it writable by the user that is running the bradge?

**Comment by altheeagle** [comment]
I've manually added the value to the config file and now i get some more information in the log. 
If I read the Log correctly there is a request from 10.201.240.80 for a description.xml. This is my FireTV, my Echo Dot has the IP-Adresse 10.201.240.40, wich I can't spot in the log.
[habridge-log2.txt](https://github.com/bwssytems/ha-bridge/files/4303284/habridge-log2.txt)

**Comment by Nachtfalke49** [comment]
I have the same problem. No device are discovered. I had to reset my Echo Dot too.

**Comment by gohamstergo** [comment]
My guess is that Amazon is screwing around again. Same as they didnt when they blocked ports other than 80. I added a new device and my harmony hub picked it up immediately, but the echo will not. Really wish Amazon would stop sucking.

**Comment by altheeagle** [comment]
@bwssytems yes, thats rigth. Its the only IPv4 Adresse of the System, there are also couple of IPv6 Adresses assigned to the Networking Device.

**Comment by joshtaz** [comment]
I am having the same issue. Harmony will pick up/control my devices, but my Echos will not.

**Comment by cylon147** [comment]
facing the same problem with my 2 different setups

**Comment by DrUdoZucker** [comment]
Hi, as my setups have now stopped too - HABridge not longer detected by Echo - it worked fine for a very long time-  I would like to know, when you will release an update (or where I can find your work in progress on this issue). Keep up your excellent work, which is highly appreciated.

**Comment by heikoh81** [comment]
Same problem here, has been working flawlessly for more than a year, now rediscovery is not working any more.

**Comment by glandecki** [comment]
Same problem. Suddenly stopped working and no success with different times nor igmp settings on switches/routers. Worked for a long time - now useless with Alexa

**Comment by gohamstergo** [comment]
> Same problem. Suddenly stopped working and no success with different times nor igmp settings on switches/routers. Worked for a long time - now useless with Alexa

To be clear, it's Amazon that is useless. They are the ones that keep changing things needlessly.

**Comment by SelmaUrban** [comment]
What is your echo firmware version?

**Comment by Nachtfalke49** [comment]
I have 3:

Cube = 17951953028
Dot 2 = 651614420
Dot 3 = 3658075524

I tested it with all three. Always pulled the power plug from the others. All don't work.

**Comment by heikoh81** [comment]
> To be clear, it's Amazon that is useless. They are the ones that keep changing things needlessly.

+1
What I don't get is how they don't break finding devices on phyiscal hue bridges out there.
I don't suppose Hue is always changing their UPNP-Timings whatsoever...

**Comment by glandecki** [comment]
Maybe it is related that Philips announced that they cancel support to v1 bridge devices and they will not work with alexa. Is it possible to emulate higher version of bridge ?
Refer to:
https://www.imore.com/philips-hue-bridge-v1-support-ending-april-2020

**Comment by heikoh81** [comment]
I don't have any physical hue bridge.
However, I always was under the impression that v1 bridge had local only support, while v2 needed a cloud connector?
I think amazon will annoy many many people if they drop local hue emulation. I think many many tuya-plugs work that way, and tasmota also emulates hue or belkin-bridges.

**Comment by audiofreak9** [comment]
> Is it possible to emulate higher version of bridge ?
As @heikoh81 stated, v1 is local discovery, v2 is cloud, so not can't emulate cloud.

**Comment by SelmaUrban** [comment]
I found this about emulating a Hue bridge v2 with local discovery (see the Security section!):
https://daenney.github.io/2019/04/09/emulating-philips-hue-bridge

The autor writes about the diyHue-Project.
Here i found the file, which creates a self-signed cert: https://github.com/mariusmotea/diyHue/blob/master/BridgeEmulator/easy_install.sh
that is used in:
https://github.com/diyhue/diyHue/blob/master/BridgeEmulator/HueEmulator3.py

@bwssytems 
If v1 local discovery (http only) was droppt, is can be a option
can you take a look at it and say my opinion on it

**Comment by Luigi8723** [comment]
Here same problem. After a update on the Echo (gen2) all devices gone. Had to reset the device and now i have some devices back. But i am still missing some.

**Comment by bwssytems** [comment]
@SelmaUrban There is no need to use the HTTPS as that is an option and not a requirement.

**Comment by glandecki** [comment]
An update - I've played with Alexa and settings and reset several times the Alexa device. It looks like amazon problem (caching ?). I noticed that trying to run discovery not from Alexa itself (but from alexa app on iphone - add device, hue bridge v1) just after resetting device worked. 
Due to the lack of documentation for amazon echo discovery protocols and caching I can't say what happened to make it work, but essentially I've reset alexa and tried discovery from device and iphone app with bridge settings with and without "link" enabled in security settings (I've cleared all existing links beforehand). No luck for several tries. I've also ran tcpdump on my emulated controller interface. I don't get it but sometimes I could see alexa attempted to read from port 80 /description.... and sometimes not. 
After a while I've noticed that alexa discovered devices, but reported an error while trying to control them.
Strange enough, habridge reported "security errors" "user not authorized" even when links were turned off. However after several tries and turning the linking off in security settings I've tried turning the emulated hue device (light)  'on/off' from the mobile app and "auto insert user" appeared in "manage links" and all works (so far), so definitely functionality is not (yet) removed from the newest Alexa devices (I have echo show 5).

**Comment by bwssytems** [comment]
For all: The new Release Candidate is out for testing.... https://github.com/bwssytems/ha-bridge/releases/tag/v5.3.1RC1

Please post your issues or successes at #1192

**Comment by ChristianGreiner** [comment]
What I did to discover devices again:
- Remove all devices + Rooms in the Alexa-App
- Renumber all devices in the web interface
- Restart the raspberry pi
- Turn off alexa and wait a few seconds and turn it on again
- Search for devices

**Comment by amyth1884** [comment]
Tried this method as well, still no luck 

> What I did to discover devices again:
> 
> * Remove all devices + Rooms in the Alexa-App
> * Renumber all devices in the web interface
> * Restart the raspberry pi
> * Turn off alexa and wait a few seconds and turn it on again
> * Search for devices
